### PR TITLE
[1.13.x] [RHPAM-4129] - Kogito Operator scorecard errors

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

Jira Issue : https://issues.redhat.com/browse/RHPAM-4129

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]
Looks ok, briefly checked

**Comment by radtriste** [comment]
@vaibhavjainwiz why such a big change to solve the scorecard errors ?

**Comment by spolti** [comment]
@radtriste mostly because the way the operator-sdk works, we had to ducplicate the missing classes to the rhpam/v1 api and update how ithose dependencies are called internally, instead referencing v1beta1 api on most places itis now calling a interface that handles better if the call is being done by the rhpam or kogito api.

File: core/kogitoinfra/infinispan_reconciler.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ const (
 var (
 	//Infinispan variables for the KogitoInfra deployed infrastructure.
 	//For Quarkus: https://quarkus.io/guides/infinispan-client#quarkus-infinispan-client_configuration
-	//For Spring: https://github.com/infinispan/infinispan-spring-boot/blob/main/infinispan-spring-boot-starter-remote/src/test/resources/test-application.properties
+	//For Spring: https://github.com/infinispan/infinispan-spring-boot/blob/master/infinispan-spring-boot-starter-remote/src/test/resources/test-application.properties
```
Comment (spolti) [inline core/kogitoinfra/infinispan_reconciler.go:51]: shouldbe main.

**Review by spolti — CHANGES_REQUESTED** [review]
just a minor thing

File: core/kogitoinfra/infinispan_reconciler.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply core/kogitoinfra/infinispan_reconciler.go:51]: done

**Review by spolti — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**